src/transformStateWithClones.js

```diff
@@ -6,8 +6,30 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let currentState = { ...state };
```

Comment:
The variable name `currentState` is good as it indicates it's a mutable copy of the state. However, make sure to push the copy of the object to the array at the end of the each loop cycle, but outside of the `switch` block, after the current action is processed. This will ensure that the same action is not performed multiple times and adheres to the DRY principle.